implementations/code_df/open_issue_age.py

```diff
@@ -0,0 +1,138 @@
+
+        self.df = self._add_open_issue_age_col(self.df)
+
+    def compute(self):
+        """
+        Compute the average open issue age for all issues in the Perceval data.
+
+        :returns avg_open_issue_ages: the average age of open
+            issues
+        """
+
+        open_issue_ages = [age for age in self.df['open_issue_age']
+                           if not pd.isna(age)]
+
+        avg_open_issue_ages = np.mean(open_issue_ages)
+        return avg_open_issue_ages
+
+    def compute_timeseries(self, period="month"):
+        """
+        The metric value is computed for each fixed interval of time
+        from the "since" date to the "until" date arguments, specified
+        during object initiation.
+        The fixed time interval can be either a month or a week.
+
+        :param period: A string which can be either "month" or "week".
+
+        :returns dataframe: A DataFrame whose rows each represent an interval
+            of "period" and the count for that interval.
+        """
+
+        if self.since is None:
+            self.since = utils.get_date(self.df, 'since')
+        if self.until is None:
+            self.until = utils.get_date(self.df, 'until')
+
+        df = self.df
+        if period == "month":
+            timeseries_series = df[['created_date', 'open_issue_age']]  \
+                .groupby([df['created_date'].dt.year.rename('year'),
+                          df['created_date'].dt.month.rename('month')]) \
+                .agg('mean')
+
+            all_periods = pd.DataFrame(
+                pd.date_range(self.since, self.until, freq='M'),
+                columns=["Dates"])
+
+            all_periods = pd.DataFrame(
+                [all_periods['Dates'].dt.year.rename("year"),
+                 all_periods['Dates'].dt.month.rename("month")]).T
+
+        elif period == "week":
+            timeseries_series = df[['created_date', 'open_issue_age']]  \
+                .groupby([df['created_date'].dt.year.rename('year'),
+                          df['created_date'].dt.week.rename('week')]) \
+                .agg('mean')
+
+            all_periods = pd.DataFrame(
+                pd.date_range(self.since, self.until, freq='W'),
+                columns=["Dates"])
+
+            all_periods = pd.DataFrame(
+                [all_periods['Dates'].dt.year.rename("year"),
+                 all_periods['Dates'].dt.week.rename("week")]).T
+
+        else:
+            raise ValueError("period parameter can take 'month' or 'week'")
+
+        timeseries_df = pd.DataFrame(timeseries_series)
+        timeseries_df.reset_index(inplace=True)
+        timeseries_df.columns = ["year", period, "open_issue_age"]
+        timeseries_df['open_issue_age'] = \
+            timeseries_df['open_issue_age'].round(decimals=2)
+
+        merged_df = all_periods.merge(timeseries_df, how='outer').fillna(0)
+
+        dataframe = merged_df
+        return dataframe
+
+    def _add_open_issue_age_col(self, df):
```

Comment:
This could be renamed to `_calculate_open_issue_age` as it actually modifying the passed dataframe by calculating it and adding it to the df. What do you think @Polaris000?

Reply:
No problem!